raw-docs/architecture.md

```diff
@@ -0,0 +1,70 @@
+
+[renjin](http://www.renjin.org/) is a JVM-based interpreter for the R language for statistical computing. Processing.R uses renjin to run R language sketches in JVM using Processing's default Java API. The Processing.R mode has a similiar architecture to [python mode](http://py.processing.org/), which uses Jython.
+
+## Architecture
+
+Processing.R is available in two forms:
+
+* The Processing.R JAR wrapper, `runner.jar`.
+* The Processing.R PDE language mode, `RLangMode`.
+
+These two forms share the same architecture.
+
+### COMMON
+
+#### `RLangPApplet`
+
+`RLangPApplet` is a class that extends `BuiltinApplet`. `RLangPApplet` overrides built-in functions such as `draw`, `settings` and `setup`. `RLangPApplet` calls renjin, which is the JVM interpreter for R, to evaluate the R code.
+
+#### `BuiltinApplet`
+
+`BuiltinApplet` is the layer that maps between the Processing Java API and the R language as implemented in renjin. For example, it defines a function to cast parameters from integer to double in order to fit the Processing API.
+
+### JAR
+
+Want to understand `runner.jar`? Start with `Runner`.
+
+#### `Runner`
+
+`Runner` is the class to run Processing.R sketches. It initializes an `RLangPApplet` instance and calls `runSketchBlocking` to run a Processing.R sketch. The entry of `runner.jar` is the `main` function in `Runner`.
+
+#### `SketchRunner`
+
+`SketchRunner` interacts with `Runner` to run the Processing.R sketches. It is the middle layer between `Runner` and `SketchServiceRunner`. 
+
+#### `SketchServiceRunner`
+
+`SketchServiceRunner` is a standalone process out of the mode. It calls the `main` function in `SketchRunner`.
+
+### PDE MODE
+
+Want to understand the Processing Development Environment(PDE) mode? Start with `RLangMode`.
+
+#### `RLangMode`
+
+`RLangMode` is the main class for the Processing.R PDE mode. It is accessed through reflection. Processing.R uses Java Remote Method Invocation(RMI), to solve the performance problem. Processing.R mode creates some long-running processes(`SketchServiceRunner`) to handle the requests from Processing Development Environment(PDE).
+
+#### `RLangEditor`
+
+`RLangEditor` is the class that represents the built-in text editor for R mode in the Processing Development Environment(PDE). It extends PDE's `Editor`, builds the mode editor, and initializes all related components, such as a formatter toolbar.
+
+#### `ModeService` and `SketchServiceManager`
+
+`ModeService` is a Remote Method Invocation(RMI) interface, which extends `java.rmi.Remote`. `SketchServiceManager` is the class which implements `ModeService`. `RLangMode` creates a `SketchServiceManager` instance when started. `SketchServiceManager` manages all `SketchServiceRunner`s, and `SketchServiceRunner` runs the real logic of Processing.R sketches.
```

Comment:
Not sure I understand. Is this sed for changing to "an Remote Method Invocation" and "an SketchServiceManager" ? Those changes aren't  needed.

"a" comes before closed consonant letters and sounds here like "Remote" and "Sketch", "an" would come before open vowel letters and sounds like "Interface" or "Extends."